File: budgeteer-api/Models/Account.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace BudgeteerAPI.Models {

    public enum AccountType {
        [EnumMember(Value = "Chequing")] CHEQUING,
        [EnumMember(Value = "Savings")] SAVINGS,
        [EnumMember(Value = "Credit Card")] CREDIT_CARD
    }

    public enum AccountBank {
        [EnumMember(Value = "CIBC (Personal)")] CIBC_PERSONAL,
        [EnumMember(Value = "PayPal (Personal)")] PAYPAL_PERSONAL,
        [EnumMember(Value = "Stack")] STACK
    }

    public class Account {
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public string UserId { get; set; } = "";

        public string Name { get; set; } = "";

        public AccountType Type { get; set; }

        public AccountBank Bank { get; set; }

        public DateTime LastImport { get; set; }

        [NotMapped]
        public long Balance { get; set; }

    }
}

File: budgeteer-web/src/models/Account.ts
export interface IAccount {
    id: string
    name: string
    balance: number
    type: AccountType
    bank: AccountBank
    lastImport: Date
}

export enum AccountBank {
    CIBC_PERSONAL = 'CIBC (Personal)',
    PAYPAL_PERSONAL = 'PayPal (Personal)',
    STACK = 'Stack',
}

export enum AccountType {
    CHEQUING = 'Chequing',
    SAVINGS = 'Savings',
    CREDIT_CARD = 'Credit Card',
}

export function getAccountImage(account: IAccount) {
    switch (account.bank) {
        case AccountBank.CIBC_PERSONAL:
            return '/img/banks/cibc-personal.svg'
        case AccountBank.PAYPAL_PERSONAL:
            return '/img/banks/paypal-personal.svg'
        case AccountBank.STACK:
            return '/img/banks/stack.png'
    }
}